src/extension.ts

```diff
@@ -151,6 +151,13 @@ export function activate(context: vscode.ExtensionContext): KafkaExtensionPartic
             ConsumerVirtualTextDocumentProvider.SCHEME, consumerVirtualTextDocumentProvider)
     );
 
+    // Refresh cluster provider participant when a vscode extension is installed/uninstalled
+    if (vscode.extensions.onDidChange) {// Theia doesn't support this API yet
+        context.subscriptions.push(vscode.extensions.onDidChange(() => {
```

Comment:
only if relevant extensions (labeled kafka-provider) changed. No need to refresh the definitions if an unrelated extension was added/removed

Reply:
Refresh a definition is just loop for package.json and get information about cluster provider. I have not found the caability to know which extension is installing, so you need to loop for all extensions and search in package.json the labeled kafka-provider. It's the same thing than collect cluster provider information.